### PR TITLE
Add anthropic model support

### PR DESCRIPTION
**Pull request by Char15Xu**
In this PR, we add support for claude 3.5 sonnet for #48 

[TODO] run `make claude`  to test if current implementation work as expected.

### CODE REVIEW

**Review by zenodflow — CHANGES_REQUESTED** [review]
.DS_Store should be killed. Btw., they should be blocked by the .gitignore.

File: Makefile [hunk 1]
```diff
@@ -153,3 +153,7 @@ duckdb:
 	python examples/apps/integration/duckdb_cluster.py
 mongodb:
 	python examples/apps/integration/mongodb_cluster.py
+
+.PHONY: claude
+claude:
+	python examples/model/claude.py
```
Comment (zenodflow) [inline Makefile:159]: https://github.com/abcsys/libem/blob/eb1f549375ee083b43cf1e71f1e3db91e39e33e7/Makefile#L107

Move between gpt and llama

File: libem/core/model/__init__.py [hunk 2]
```diff
@@ -15,6 +15,8 @@ async def async_call(*args, **kwargs) -> dict:
         return llama.call(*args, **kwargs)
```
Comment (zenodflow) [inline libem/core/model/__init__.py:15]: match kwargs.get("model", ""):

case ... : 
case ... : 
case _

File: libem/core/model/claude.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    libem.LIBEM_CONFIG.get("CLAUDE_API_KEY", "")
+)
+
+_client = None
+
+
+def get_client():
+    global _client
+
+    if not os.environ.get("CLAUDE_API_KEY"):
```
Comment (zenodflow) [inline libem/core/model/claude.py:25]: Nice. This should be able to configure in the libem.cli

**Comment by zenodflow** [comment]
@Char15Xu Should we add anthropic package to requirements.txt?

**Comment by Char15Xu** [comment]
> @Char15Xu Should we add anthropic package to requirements.txt?

will do

**Comment by zenodflow** [comment]
@Char15Xu Nice. Let me know if this is ready for another review.

**Comment by Char15Xu** [comment]
@zenodflow Yes, this is ready for review

File: cli/libem [hunk 4]
```diff
@@ -82,13 +82,23 @@ def configure(args):
...
+    new_openai_key = input(f"Enter OPENAI_API_KEY (press Enter to keep the existing key: "
+                    f"'{mask_key(existing_openai_key)}'): ").strip()
 
     # If no input, keep the existing key; otherwise, update
-    if new_key:
-        config['OPENAI_API_KEY'] = new_key
+    if new_openai_key:
+        config['OPENAI_API_KEY'] = new_openai_key
+    else:
+        print("No input provided. Using existing API key.")
```
Comment (zenodflow) [inline cli/libem:93]: Remove this print

File: cli/libem [hunk 5]
```diff
@@ -82,13 +82,23 @@ def configure(args):
...
+
+    existing_claude_key = config.get('CLAUDE_API_KEY', '')
+    new_claude_key = input(f"Enter CLAUDE_API_KEY (press Enter to keep the existing key: "
+                    f"'{mask_key(existing_claude_key)}'): ").strip()
+
+    # If no input, keep the existing key; otherwise, update
+    if new_claude_key:
+        config['CLAUDE_API_KEY'] = new_claude_key
     else:
         print("No input provided. Using existing API key.")
```
Comment (zenodflow) [inline cli/libem:103]: Remove this print

**Review by zenodflow — APPROVED** [review]
One last nit. But overall LGTM. Nice work!

**Comment by Char15Xu** [comment]
libem/optimize/price.json already include information on Claude model with the correct naming. Once the PR is approved, i will move on to test anthropic model on abt-buy dataset.

**Pull request merged**